siliconcompiler/core.py

```diff
@@ -3839,7 +3811,7 @@ def _runtask(self, step, index, status):
         ##################
         # Run preprocess step for tool
         if not is_builtin:
-            func = self.find_function(tool, "pre_process", 'tools', task)
+            func = getattr(self._get_task_module(step, index, flow=flow), 'pre_process', None)
```

Comment:
This is why I think it'll be great to move to using Python modules.

However, I think that this sort of logic might be inefficient, because it will fully re-import the module every time that the `_get_task_module` is called.

Could we cache loaded modules somewhere to mitigate that?